Create initial awesome-stateless list with curated stateless web applications (#2)
revolter/awesome-stateless: README.md

- **Share-by-URL apps** - Tools where you can share content just by sending the link
- **Database-less apps** - Applications that don't persist data server-side
- **Self-contained apps** - Applications where the URL itself contains all the data

The pattern is commonly used in text editors, JSON visualizers, regex testers, drawing tools, and other utilities where users want to quickly share their work without creating accounts or saving files.

## Contents

- [Text Editors & Note Taking](#text-editors--note-taking)
- [Development Tools](#development-tools)
- [Utilities](#utilities)
- [Diagramming & Visualization](#diagramming--visualization)

## Text Editors & Note Taking

- [Notepad Tab](https://notepadtab.com) - A simple, secure and private note taking tool. *Disclaimer: Owned by the repository maintainer.*

## Development Tools

- [Diff Viewer](https://diffviewer.vercel.app) - A privacy focused, easy sharable, open source and anonymous tracking diff viewer.

## Utilities

- [IBAN Advertiser](https://iban.ad) - The easiest way for NGOs, freelancers, and clubs to share their bank details for donations, payments, or memberships. *Disclaimer: Owned by the repository maintainer.*

> - [Carbon](https://carbon.now.sh) - Create and share beautiful images of your source code with all styling encoded in the URL.

Reviewer: @copilot, remove the superfluous explanation from the end of the description:
```suggestion
- [Carbon](https://carbon.now.sh) - Create and share beautiful images of your source code.
```

## Diagramming & Visualization

- [Mermaid](https://mermaid.live) - JavaScript based diagramming and charting tool that renders Markdown-inspired text definitions to create and modify diagrams dynamically.

> - [draw.io](https://app.diagrams.net) - Online diagramming tool that can encode diagrams in the URL for sharing.

Reviewer: @copilot, remove this tool as it's not stateless

These applications are also known by various names:
- **URL-encoded state apps** - Applications that serialize their state directly into the URL
- **Share-by-URL apps** - Tools where you can share content just by sending the link
- **Database-less apps** - Applications that don't persist data server-side
- **Self-contained apps** - Applications where the URL itself contains all the data

The pattern is commonly used in text editors, JSON visualizers, regex testers, drawing tools, and other utilities where users want to quickly share their work without creating accounts or saving files.

## Contents

- [Text Editors & Note Taking](#text-editors--note-taking)
- [Development Tools](#development-tools)
- [Utilities](#utilities)
- [Diagramming & Visualization](#diagramming--visualization)

## Text Editors & Note Taking

- [Notepad Tab](https://notepadtab.com) - A simple, secure and private note taking tool. *Disclaimer: Owned by the repository maintainer.*

## Development Tools

- [Diff Viewer](https://diffviewer.vercel.app) - A privacy focused, easy sharable, open source and anonymous tracking diff viewer.

> - [regex101.com](https://regex101.com) - Online regex tester that encodes regex patterns, test strings, and flags in the URL.

Reviewer: @copilot, remove this tool as it isn't stateless

## What are Stateless Web Applications?

Stateless web applications encode their entire state or user-generated content directly in the URL instead of storing it in a database. This approach enables instant sharing, eliminates the need for user accounts, and makes the applications completely self-contained.

These applications are also known by various names:
- **URL-encoded state apps** - Applications that serialize their state directly into the URL
- **Share-by-URL apps** - Tools where you can share content just by sending the link
- **Database-less apps** - Applications that don't persist data server-side
- **Self-contained apps** - Applications where the URL itself contains all the data

The pattern is commonly used in text editors, JSON visualizers, regex testers, drawing tools, and other utilities where users want to quickly share their work without creating accounts or saving files.

## Contents

- [Text Editors & Note Taking](#text-editors--note-taking)
- [Development Tools](#development-tools)
- [Utilities](#utilities)
- [Diagramming & Visualization](#diagramming--visualization)

## Text Editors & Note Taking

> - [notepadtab.com](https://notepadtab.com) - A simple, secure and private note taking tool. *Disclaimer: Owned by the repository maintainer.*

Reviewer: @copilot, use the official name for the link:
```suggestion
- [Notepad Tab](https://notepadtab.com) - A simple, secure and private note taking tool. *Disclaimer: Owned by the repository maintainer.*
```

## Development Tools

> - [diffviewer.vercel.app](https://diffviewer.vercel.app) - Side-by-side diff viewer that encodes both texts in the URL for easy sharing.

Reviewer: @copilot, use the official name and description:
```suggestion
- [Diff Viewer](https://diffviewer.vercel.app) - A privacy focused, easy sharable, open source and anonymous tracking diff viewer.
```
